### PR TITLE
ENH: add a setter to the epics motor interface limits attribute.

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Add a setter to the epics motor interface limits attribute.

## Motivation and Context

XCS was having a problem with this line in `LookupTablePositioner`:
https://github.com/pcdshub/pcdsdevices/blob/af187b6b26235f95056cf9b168bb6decf6627c4c/pcdsdevices/pseudopos.py#L990
Where it sets the limits of the real motor.

Rather than consider ways that this could be done differently to avoid EPICS puts on init, I opted to add the missing limits setter as a quick fix.

## How Has This Been Tested?

Live in XCS

## Where Has This Been Documented?

Release notes only

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: pcdsdevices/epics_motor.py [hunk 1]
```diff
@@ -177,17 +177,27 @@ def format_status_info(self, status_info):
...
         return limits
 
+    @limits.setter
+    def limits(self, lims: tuple[float, float]):
+        """
+        Write to the epics limits on setattr
+
+        Expects a tuple of two elements, the low and high limits to set.
+        """
+        self.low_limit_travel.put(min(lims))
```
Comment (klauer) [inline pcdsdevices/epics_motor.py:198]: Maybe `if not len(lims) == 2: raise ValueError(...)`

And _maybe_ have special handling for `if min(lims) == max(lims)` ? That implies "disable limits or low, high = (0, 0)" in the motor record world I believe

File: pcdsdevices/epics_motor.py [hunk 2]
```diff
@@ -177,17 +177,27 @@ def format_status_info(self, status_info):
...
 
-    def _get_epics_limits(self):
+    def _get_epics_limits(self) -> tuple[float, float]:
         limits = self.user_setpoint.limits
         if limits is None or limits == (None, None):
             # Not initialized
             return (0, 0)
         return limits
 
+    @limits.setter
```
Comment (klauer) [inline pcdsdevices/epics_motor.py:191]: Setters that put to PVs have been divisive in the past...
I think in this case the utility of it is justified and am on board

**Review by klauer — APPROVED** [review]

File: pcdsdevices/epics_motor.py [hunk 3]
```diff
@@ -177,17 +177,27 @@ def format_status_info(self, status_info):
...
+    def limits(self, lims: tuple[float, float]):
+        """
+        Write to the epics limits on setattr
+
+        Expects a tuple of two elements, the low and high limits to set.
+        """
+        self.low_limit_travel.put(min(lims))
+        self.high_limit_travel.put(max(lims))
```
Comment (tangkong) [inline pcdsdevices/epics_motor.py:199]: I know this is an entirely logical assumption to make, but are we sure we can be certain of this?  We don't have any whacky motors that have the limits switched and would require the order to matter?

(a genuine question, I could see some zany IOC that swaps things around for some unreasonable reason)

Maybe my imagination is getting the better of me

**Review by tangkong — APPROVED** [review]
LGTM despite my harebrained comment

File: pcdsdevices/epics_motor.py (hunk 3 above)
Comment (klauer) [reply pcdsdevices/epics_motor.py:199]: That line of questioning reminds me - it won't be a problem for `high_limit_travel < low_limit_travel` on the IOC side as you've implied but there _is_ a potential issue there:
You can't set a high-limit below the low limit, and you can't set the low limit above the current high limit. That needs some special handling to my recollection.

File: pcdsdevices/epics_motor.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsdevices/epics_motor.py:198]: Both of these are worth doing

File: pcdsdevices/epics_motor.py (hunk 2 above)
Comment (ZLLentz) [reply pcdsdevices/epics_motor.py:191]: I agree, it's a lot less annoying than requiring some other method that every motor-like needs for changing the limits value.

File: pcdsdevices/epics_motor.py (hunk 3 above)
Comment (ZLLentz) [reply pcdsdevices/epics_motor.py:199]: I think we're guaranteed this by the way the IOC works, but it's a good question

Comment (ZLLentz) [reply pcdsdevices/epics_motor.py:199]: oh no you're so right Ken, there's some awful stateful behavior here depending on the original limit settings

**Comment by ZLLentz** [comment]
This became a bit more involved so I added some unit tests too

File: pcdsdevices/epics_motor.py (hunk 3 above)
Comment (klauer) [reply pcdsdevices/epics_motor.py]: Glad I commented then, despite the haziness... Fix looks good, I think

**Review by klauer — APPROVED** [review]
Updated fix LGTM - bonus points for the nice test

**Review by tangkong — APPROVED** [review]
LGTM 👍

**Pull request merged**